src/cascadia/TerminalSettingsModel/Profile.cpp

```diff
@@ -331,50 +323,22 @@ void Profile::LayerJson(const Json::Value& json)
     JsonUtils::GetValueForKey(json, StartingDirectoryKey, _StartingDirectory);
 
     JsonUtils::GetValueForKey(json, IconKey, _Icon);
-    JsonUtils::GetValueForKey(json, BackgroundImageKey, _BackgroundImagePath);
-    JsonUtils::GetValueForKey(json, BackgroundImageOpacityKey, _BackgroundImageOpacity);
-    JsonUtils::GetValueForKey(json, BackgroundImageStretchModeKey, _BackgroundImageStretchMode);
-    JsonUtils::GetValueForKey(json, BackgroundImageAlignmentKey, _BackgroundImageAlignment);
-    JsonUtils::GetValueForKey(json, RetroTerminalEffectKey, _RetroTerminalEffect);
     JsonUtils::GetValueForKey(json, AntialiasingModeKey, _AntialiasingMode);
     JsonUtils::GetValueForKey(json, TabColorKey, _TabColor);
     JsonUtils::GetValueForKey(json, BellStyleKey, _BellStyle);
-    JsonUtils::GetValueForKey(json, PixelShaderPathKey, _PixelShaderPath);
-}
 
-// Method Description:
-// - Either Returns this profile's background image path, if one is set, expanding
-// - Returns this profile's background image path, if one is set, expanding
-//   any environment variables in the path, if there are any.
-// - Or if "DesktopWallpaper" is set, then gets the path to the desktops wallpaper.
-// Return Value:
-// - This profile's expanded background image path / desktops's wallpaper path /the empty string.
-winrt::hstring Profile::ExpandedBackgroundImagePath() const
-{
-    const auto path{ BackgroundImagePath() };
-    if (path.empty())
-    {
-        return path;
-    }
-    // checks if the user would like to copy their desktop wallpaper
-    // if so, replaces the path with the desktop wallpaper's path
-    else if (path == DesktopWallpaperEnum)
+    if (json.isMember(JsonKey(UnfocusedAppearanceKey)))
     {
-        WCHAR desktopWallpaper[MAX_PATH];
+        auto unfocusedAppearance{ winrt::make_self<implementation::AppearanceConfig>() };
 
-        // "The returned string will not exceed MAX_PATH characters" as of 2020
-        if (SystemParametersInfo(SPI_GETDESKWALLPAPER, MAX_PATH, desktopWallpaper, SPIF_UPDATEINIFILE))
-        {
-            return winrt::hstring{ (desktopWallpaper) };
-        }
-        else
-        {
-            return winrt::hstring{ L"" };
-        }
-    }
-    else
-    {
-        return winrt::hstring{ wil::ExpandEnvironmentStringsW<std::wstring>(path.c_str()) };
+        // If an unfocused appearance is defined in this profile, any undefined parameters are
+        // taken from this profile's default appearance, so add it as a parent
+        com_ptr<AppearanceConfig> parentCom;
+        parentCom.copy_from(defaultAppearanceImpl);
+        unfocusedAppearance->InsertParent(parentCom);

```

Comment:
See my comment for my review. This shows a bit of what I'm talking about.

Here, we're having out `unfocusedAppearance` inherit from our `defaultAppearance`. This works for now because we're handling this when we layer the json, but it won't work if we enter a world where we want to interact with the settings model in a purely object-oriented approach. This is because we would need somebody using our API to manually create an inheritance relationship between `unfocusedAppearance` and `defaultAppearance` every time.

If we go with my proposal in the main review comment, we won't have to deal with that. The inheritance will always be preserved because it's the `Profile`'s job to figure out what the value is, not the `AppearanceConfig`'s.

Hope this helps.

Reply:
Resolving this since we discussed it offline!